app/src/main/java/org/oppia/android/app/profile/AdminSettingsDialogFragmentPresenter.kt

```diff
@@ -42,12 +42,17 @@ class AdminSettingsDialogFragmentPresenter @Inject constructor(
     }
 
     binding.adminSettingsInputPinEditText.setText(adminViewModel.inputPin.get().toString())
+    var tempError = adminViewModel.errorMessage.get()
 
     // [onTextChanged] is a extension function defined at [TextInputEditTextHelper]
     binding.adminSettingsInputPinEditText.onTextChanged { confirmPin ->
       confirmPin?.let {
         adminViewModel.inputPin.set(confirmPin.toString())
-        adminViewModel.errorMessage.set("")
+        if (tempError != "") {
+          tempError = ""
+        } else {
+          adminViewModel.errorMessage.set("")
+        }
```

Comment:
@anandwana001 I've added this cause after the configuration change the errorMessage is set blank
Please check if it is a good practice for this

Reply:
Could you please describe the issue and how this is fixing? If this is something logical, let's create a good first issue for this.

Reply:
@anandwana001 the problem here is after rotating to landscape the `onTextChanged` is being called and the errorMessage is set to empty string.

So I created a temporary variable and stored the errorMessage in it 
and I'm skipping `adminViewModel.errorMessage.set("")` if the tempError has any value 
here the tempError is only created when the fragment starts

so that after the orientation change the errorMessage does'nt disappear

Reply:
Can we do this through onRestoreInstance, take reference from admin auth screen?

Reply:
@anandwana  Here the value is stored in the viewmodel but it gets changed inside onTextChanged 

We can use onRestoreInstance here as it would be another variable we can set the value for the viewmodel later 

But I think this is not necessary as we already have the error message in the viewmodel

Also I've checked the onRestoreInstance in AdminAuth and it works without onRestoreInstance too as the value is saved in - the viewmodel

So shall I do use this method or implement onRestoreInstance for this

Reply:
@anandwana001  Here the value is stored in the viewmodel but it gets changed inside onTextChanged 

We can use onRestoreInstance here as it would be another variable we can set the value for the viewmodel later 

But I think this is not necessary as we already have the error message in the viewmodel

Also I've checked the onRestoreInstance in AdminAuth and it works without onRestoreInstance too as the value is saved in - the viewmodel

So shall I do use this method or implement onRestoreInstance for this

Reply:
@anandwana001 As this was a bit complicated
I've made some changes suggested by rajat can you please review it

Reply:
Do we have similar issue with other parts of migration? If yes, we need to fix them too and if it we need to check what causing this issue here only!!

Reply:
No, I think this is because the fragment is being recreated and then the onTextChanged is triggered as it is a fragmentDialog

Reply:
I am not much inclined with this solution. Looking into what other option we can have.

Reply:
Ok, Shall i try using onSavedInstance here?